### PR TITLE
Fix regex for STP downloads

### LINKED ISSUE #34875: Safari Technology Preview downloads are failing after page update
[Pipeline run](https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=84583&view=logs&jobId=d41515ed-d71e-5e8d-de88-940c2a934d56&j=42513d51-f539-5b97-080a-60c327907e2e&t=0831e07d-b569-54b4-d687-33f1efddc82e)

```sh
Traceback (most recent call last):
  File "/Users/runner/work/1/s/./wpt", line 10, in <module>
    wpt.main()
  File "/Users/runner/work/1/s/tools/wpt/wpt.py", line 220, in main
    rv = script(*args, **kwargs)
  File "/Users/runner/work/1/s/tools/wpt/install.py", line 89, in run
    install(browser, kwargs["component"], destination, channel, logger=logger,
  File "/Users/runner/work/1/s/tools/wpt/install.py", line 109, in install
    path = getattr(browser_cls(logger), method)(dest=destination, channel=channel, **kwargs)
  File "/Users/runner/work/1/s/tools/wpt/browser.py", line 1612, in download
    image_path = self._download_image(stp_downloads, tmpdir, system_version)
  File "/Users/runner/work/1/s/tools/wpt/browser.py", line 1530, in _download_image
    raise ValueError(f"no download for {system_version}")
ValueError: no download for 12.4

```

Similar to #34752 

Looks like the page has changed.
Here's a diff between the [file created for testing ](https://github.com/web-platform-tests/wpt/blob/master/tools/wpt/tests/safari_downloads_page_stp_section.html) and now

```diff
 <div class="column large-6 medium-12 small-12">
     <div class="callout">
-        <figure class="app-icon large-icon safari-preview-icon" aria-hidden="true" data-hires-status="pending"></figure>
+        <figure class="app-icon large-icon safari-preview-icon" aria-hidden="true"></figure>
         <h4>Safari Technology Preview</h4>
         <p class="margin-bottom-small">Get a sneak peek at upcoming web technologies in macOS and iOS with <a href="/safari/technology-preview/" class="nowrap">Safari Technology Preview</a> and experiment with these technologies in your websites and extensions.</p>
         <ul class="links small">
-            <li class="dmg" data-hires-status="pending"><a class="inline" href="https://secure-appldnld.apple.com/STP/012-38225-20220706-237860CD-5766-4F53-AAC7-1CE26023A959/SafariTechnologyPreview.dmg">Safari Technology Preview<br>for macOS&nbsp;Ventura</a><br><span class="smaller lighter nowrap nowrap-small">Requires macOS&nbsp;13 beta&nbsp;3 or later.</span></li>
-            <li class="dmg margin-top-small" data-hires-status="pending"><a class="inline" href="https://secure-appldnld.apple.com/STP/012-32918-20220629-B3452905-0138-4CA9-A4E6-334B63585653/SafariTechnologyPreview.dmg">Safari Technology Preview<br>for macOS&nbsp;Monterey</a><br><span class="smaller lighter">Requires macOS&nbsp;12.3 or later.</span></li>
-            <li class="document margin-top-small" data-hires-status="pending"><a href="/safari/technology-preview/release-notes/">Release Notes</a></li>
+            <li class="dmg"><a class="inline" href="https://secure-appldnld.apple.com/STP/012-38225-20220706-237860CD-5766-4F53-AAC7-1CE26023A959/SafariTechnologyPreview.dmg">Safari Technology Preview 148<br />for macOS&nbsp;Ventura</a><br /><span class="smaller lighter nowrap nowrap-small">Requires macOS&nbsp;13 beta&nbsp;3 or later.</span></li>
+            <li class="dmg margin-top-small"><a class="inline" href="https://secure-appldnld.apple.com/STP/012-38074-20220715-4851625A-8AC7-4155-84C0-AEB4B88F08FE/SafariTechnologyPreview.dmg">Safari Technology Preview 149<br />for macOS&nbsp;Monterey</a><br /><span class="smaller lighter">Requires macOS&nbsp;12.3 or later.</span></li>
+            <li class="document margin-top-small"><a href="/safari/technology-preview/release-notes/">Release Notes</a></li>
         </ul>
         <div class="row gutter text-left">
             <div class="column">
                 <p class="sosumi no-margin-bottom margin-top-small">Release</p>
-                <p class="smaller lighter no-margin">148</p>
+                <p class="smaller lighter no-margin">149</p>
             </div>
             <div class="column">
                 <p class="sosumi no-margin-bottom margin-top-small">Posted</p>
-                <p class="smaller lighter no-margin">June 29, 2022</p>
+                <p class="smaller lighter no-margin">July 14, 2022</p>
             </div>
         </div>
     </div>
```

@gsnedders

### PR DESCRIPTION
**Pull request by jcscottiii**
On July 14, 2022, the page was updated.

Example of STP line item previously:
`Safari Technology Preview<br>for macOS&nbsp;Ventura`

Example of STP line item now:
`Safari Technology Preview 148<br />for macOS&nbsp;Ventura`

Changes:
- Change regex to work with the format while maintaining backwards
  compatibility in case the old format comes back
- Add a test case to cover the new format
- Reformat tests to loop through the various test data files to test
  both new and previous formats

Fixes #34875

### CODE REVIEW

**Comment by jcscottiii** [comment]
This change does fix the STP download. This [run](https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=84611&view=logs&jobId=bf8f5026-3d1f-5955-26be-17f9e1ef0bb3&j=bf8f5026-3d1f-5955-26be-17f9e1ef0bb3&t=fb374013-b51a-5838-c2f2-591e11e493ab)  shows that it is able to download the right version.


However, I am trying to figure out why there are other checks still failing. 


- [x] Azure Pipelines (affected tests without changes: Safari Technology Preview)
  - Is expected to fail because it does not use my changes
- [ ] Azure Pipelines (infrastructure/ tests: macOS)
  - Not sure why it is failing. It is currently failing on the `Run tests (Firefox Nightly)` [step](https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=84605&view=logs&jobId=ca1e4672-2834-5daa-aa0e-eeaa3309447b&j=ca1e4672-2834-5daa-aa0e-eeaa3309447b&t=cfdcdcea-19e9-567e-6b35-b456d76ee34a). Which should be unrelated
- [ ] Azure Pipelines - This is the high level pipeline check. The other two Azure Pipelines need to succed
- [ ] infrastructure/ tests
  - Looks like it may be unrelated. The [logs](https://community-tc.services.mozilla.com/tasks/TImcSG8nRhigLhrnxY5fRg/runs/0/logs/public/logs/live.log) for the run are throwing exceptions from files I did not touch.
  - ```
     1:44.89 TEST_START: /infrastructure/testdriver/actions/penPointerEvents.html
     1:45.08 INFO {'actions': [{'type': 'none', 'actions': [{'type': 'pause', 'duration': 16}, {'type': 'pause', 'duration': 16}, {'type': 'pause', 'duration': 16}, {'type': 'pause', 'duration': 16}, {'type': 'pause', 'duration': 16}, {'type': 'pause', 'duration': 16}, {'type': 'pause', 'duration': 16}, {'type': 'pause', 'duration': 16}], 'id': '0'}, {'type': 'pointer', 'actions': [{'type': 'pointerMove', 'x': 0, 'y': 0, 'origin': 'viewport'}, {'type': 'pointerMove', 'x': 10, 'y': 10, 'origin': 'viewport'}, {'type': 'pointerMove', 'x': 0, 'y': 0, 'origin': {'element-6066-11e4-a52e-4f735466cecf': '5d2205ec-2a40-4dab-9b8e-3d04dc2cf5a4', 'chromeelement-9fc5-4b51-a3c8-01716eedeb04': '5d2205ec-2a40-4dab-9b8e-3d04dc2cf5a4'}}, {'type': 'pointerDown', 'button': 0}, {'type': 'pointerMove', 'x': 15, 'y': 0, 'origin': {'element-6066-11e4-a52e-4f735466cecf': '5d2205ec-2a40-4dab-9b8e-3d04dc2cf5a4', 'chromeelement-9fc5-4b51-a3c8-01716eedeb04': '5d2205ec-2a40-4dab-9b8e-3d04dc2cf5a4'}}, {'type': 'pointerMove', 'x': 30, 'y': 0, 'origin': {'element-6066-11e4-a52e-4f735466cecf': '5d2205ec-2a40-4dab-9b8e-3d04dc2cf5a4', 'chromeelement-9fc5-4b51-a3c8-01716eedeb04': '5d2205ec-2a40-4dab-9b8e-3d04dc2cf5a4'}}, {'type': 'pointerUp', 'button': 0}, {'type': 'pointerMove', 'x': 0, 'y': 0, 'origin': 'viewport'}], 'parameters': {'pointerType': 'pen'}, 'id': 'penPointer1'}]}
     1:45.13 WARNING Action action_sequence failed
     1:45.13 WARNING Traceback (most recent call last):
      File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/base.py", line 740, in process_action
        result = action_handler(payload)
      File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/actions.py", line 87, in __call__
        self.protocol.action_sequence.send_actions({"actions": actions})
      File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/executormarionette.py", line 482, in send_actions
        self.marionette._send_message("WebDriver:PerformActions", actions)
      File "/home/test/web-platform-tests/_venv3/lib/python3.8/site-packages/marionette_driver/decorators.py", line 26, in _
        return func(*args, **kwargs)
      File "/home/test/web-platform-tests/_venv3/lib/python3.8/site-packages/marionette_driver/marionette.py", line 602, in _send_message
        self._handle_error(err)
      File "/home/test/web-platform-tests/_venv3/lib/python3.8/site-packages/marionette_driver/marionette.py", line 622, in _handle_error
        raise errors.lookup(error)(message, stacktrace=stacktrace)
    marionette_driver.errors.UnknownException: Error: Unimplemented pointerMove for pointerType pen
    stacktrace:
	    pointerMove@chrome://remote/content/marionette/action.js:1491:11
	    performPointerMoveStep@chrome://remote/content/marionette/action.js:1033:25
	    dispatch/<@chrome://remote/content/marionette/action.js:1006:22
	    moveOverTime@chrome://remote/content/marionette/action.js:1427:11
     1:45.13 WARNING Traceback (most recent call last):
      File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/executormarionette.py", line 822, in run_func
        self.result = True, self.func(self.protocol, self.url, self.timeout)
      File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/executormarionette.py", line 950, in do_testharness
        done, rv = handler(result)
      File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/base.py", line 724, in __call__
        return callback(url, payload)
      File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/base.py", line 740, in process_action
        result = action_handler(payload)
      File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/actions.py", line 87, in __call__
        self.protocol.action_sequence.send_actions({"actions": actions})
      File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/executormarionette.py", line 482, in send_actions
        self.marionette._send_message("WebDriver:PerformActions", actions)
      File "/home/test/web-platform-tests/_venv3/lib/python3.8/site-packages/marionette_driver/decorators.py", line 26, in _
        return func(*args, **kwargs)
    ```
  - Also seeing some times when there are differences
  - ```
      0:28.34 TEST_START: /infrastructure/reftest/reftest_mismatch_fail-print.html
     0:28.34 pid:1677 1658164928027	Marionette	INFO	Testing http://web-platform.test:8000/infrastructure/reftest/reftest_mismatch_fail-print.html != http://web-platform.test:8000/infrastructure/reftest/reftest_match-print-ref.html
     0:30.98 pid:1677 1658164930661	Marionette	INFO	Rendering page 1/2
     0:31.04 pid:1677 1658164930721	Marionette	INFO	Rendering page 1/2
     0:31.09 pid:1677 1658164930775	Marionette	INFO	No differences allowed
     0:31.09 pid:1677 1658164930776	Marionette	INFO	Rendering page 2/2
     0:31.10 pid:1677 1658164930787	Marionette	INFO	Rendering page 2/2
     0:31.12 pid:1677 1658164930806	Marionette	INFO	No differences allowed
     0:31.12 TEST_END: FAIL
    ```
- [ ] sink-task
  - Is the higher level check. failed because `infrastructure/tests` failed

I think the outstanding failures are unrelated. But doing some experimentation on this [PR](https://github.com/web-platform-tests/wpt/pull/34881) to verify.

Would you happen to have any ideas @dbaron ?

**Review by dbaron — APPROVED** [review]
This is way out of my area, but I'm going to review+approve it anyway because I'd like to see WPT PRs get unblocked sooner rather than later.  (I reviewed one this morning and it can't currently land...)

If you were waiting for review from someone else in particular, feel free to wait on them as well.

**Comment by gsnedders** [comment]
Filed https://github.com/web-platform-tests/wpt/issues/34888 for the Firefox failures.

**Pull request closed without merging**
